### PR TITLE
feat: add support for image pull secrets for all components

### PR DESCRIPTION
**Pull request by knrc**
This pull request adds support for image pull secrets, with the ability to specify them at the top level (inherited by all) or within individual components.

## Summary by Sourcery

Add support for image pull secrets across all operator components

New Features:
- Introduce ImagePullSecrets field in CRD API types and schemas for top-level and individual components (CTlog, Fulcio, Rekor, Trillian, TimestampAuthority, TUF, Securesign)
- Add WithImagePullSecrets option in RBAC actions to propagate imagePullSecrets into ServiceAccount resources

Enhancements:
- Implement MergeImagePullSecrets utility to combine and deduplicate global and component-specific secret lists
- Update controllers to merge global and per-component ImagePullSecrets when ensuring sub-resources

Tests:
- Add unit tests for MergeImagePullSecrets covering edge and deduplication cases
- Add RBAC action tests to validate ServiceAccount imagePullSecrets behavior on create and update

### CODE REVIEW

**Review by osmman — CHANGES_REQUESTED** [review]

File: api/v1alpha1/securesign_types.go [hunk 1]
```diff
@@ -35,6 +35,8 @@ type SecuresignSpec struct {
 	Tuf                TufSpec                 `json:"tuf,omitempty"`
 	Ctlog              CTlogSpec               `json:"ctlog,omitempty"`
 	TimestampAuthority *TimestampAuthoritySpec `json:"tsa,omitempty"`
+
+	ServiceAccountRequirements `json:",inline"`
```
[the remark this answers is not in the document]
Comment (knrc) [reply api/v1alpha1/securesign_types.go]: Yes to documentation, I've already reached out to Aron about creating the docs issues.  Or are you referring to having a comment?

I'll add some tests

[the remark this answers is not in the document]
Comment (knrc) [reply api/v1alpha1/securesign_types.go]: okay, I can waive them.

File: internal/action/rbac/action.go [hunk 2]
```diff
@@ -48,10 +54,11 @@ func NewAction[T apis.ConditionsAwareObject](componentName, rbacName string, opt
...
 	action.BaseAction
-	componentName string
-	rbacName      string
-	rules         []rbacv1.PolicyRule
-	canHandle     func(context.Context, T) bool
+	componentName    string
+	rbacName         string
+	rules            []rbacv1.PolicyRule
+	canHandle        func(context.Context, T) bool
+	imagePullSecrets func(context.Context, T) []v1.LocalObjectReference
```
[the remark this answers is not in the document]
Comment (knrc) [reply internal/action/rbac/action.go]: @osmman I decided just to remove the context, as you say this can be added back in future if necessary.

**Review by osmman — CHANGES_REQUESTED** [review]

**Comment by knrc** [comment]
Looks like this is being duplicated by #1927, when this one should have been pursued

**Comment by osmman** [comment]
Closing PR, resolved in V1 API https://github.com/securesign/secure-sign-operator/pull/1927
